### PR TITLE
[F] Update hero section markup

### PR DESCRIPTION
**Pull request by naomiyaki**
[Delivers #501]
Update project hero with simplified maker listings and add draft label

### CODE REVIEW

**Comment by zdavis** [comment]
Is the gray background correct? Seems too light:
image

Spacing between desc. and button seems too tight; buttons seem less tight (taller?) than comps:
image

Buy button text isn't right:
image

Instagram icon doesn't match comps:
image

No italics on contributor name in comps:
image

Too much bottom padding on hero compared to comps:
image

Mobile line height issues on authors:
image

I realize typography is a moving target, but I wonder if the size of the desc. text is a bit too large. Thoughts?

File: client/src/components/frontend/Project/Hero.js [hunk 1]
```diff
@@ -14,17 +14,78 @@ export default class ProjectHero extends Component {
...
-        </section>
+        </div>
       );
     }
+    return (
+      <div className="project-creator-list">
+        <ul>
+          {creators.map(creator => {
+            return (
+              <li key={creator.id} maker={creator}>
```
Comment (zdavis) [inline client/src/components/frontend/Project/Hero.js]: This will throw a warning in the console because you can't have a maker prop on the `li` component. Why is this here?

File: client/src/components/frontend/Project/Hero.js [hunk 2]
```diff
@@ -14,17 +14,78 @@ export default class ProjectHero extends Component {
     project: PropTypes.object
   };
 
-  listMakers() {
+  haveAvatars(creators, variant) {
```
Comment (zdavis) [inline client/src/components/frontend/Project/Hero.js]: I'd rename this method to "hasAvatars"

File: client/src/components/frontend/Project/Hero.js [hunk 3]
```diff
@@ -14,17 +14,78 @@ export default class ProjectHero extends Component {
...
+    });
+
+    // If none of the creators had a null avatar
+    return result;
+  }
+
+  listCreators() {
     const creators = get(this.props.project, "relationships.creators");
-    if (creators && creators.length > 0) {
+    if (!creators || creators.length <= 0) return null;
```
Comment (zdavis) [inline client/src/components/frontend/Project/Hero.js]: Seems like this line of code could be abstracted into a `hasCreators` method.

File: client/src/components/frontend/Project/Hero.js [hunk 4]
```diff
@@ -14,17 +14,78 @@ export default class ProjectHero extends Component {
...
+            );
+          })}
+        </ul>
+      </div>
+    );
+  }
+
+  listContributors() {
+    const contributors = get(this.props.project, "relationships.contributors");
+    if (!contributors || contributors.length <= 0) return null;
```
Comment (zdavis) [inline client/src/components/frontend/Project/Hero.js]: As above, seems like this simple check could be abstracted into a method: `hasPeople(type = "creators")` or something like that.

File: client/src/components/frontend/Project/Hero.js [hunk 5]
```diff
@@ -14,17 +14,78 @@ export default class ProjectHero extends Component {
...
+    return (
+      <div className="project-contributor-list">
+        <span className="label">
+          {"Contributors: "}
+        </span>
+        <ul>
+          {contributors.map(creator => {
+            return (
+              <li key={creator.id} maker={creator}>
```
Comment (zdavis) [inline client/src/components/frontend/Project/Hero.js]: This will throw a warning in the console because maker is an invalid prop for `li`

**Comment by zdavis** [comment]
@naomiyaki Could you please prioritize completing this PR? I'd like this to be wrapped up before you begin any new work (eg, mobile resources).

**Comment by naomiyaki** [comment]
On it!

**Comment by naomiyaki** [comment]
@zdavis I did a force push that I believe addresses all of these issues except the Instagram icon. The icon we're using on the front-end is based off of Instagram branding changes that occurred after Lael made the original comps. I believe we had a conversation with Lael at some point about using this one instead.
I think the icon in the comps is clear, but I'm generally inclined to use official brand guidelines since it fits in just fine with the Manifold design.

**Comment by zdavis** [comment]
A number of my comments above weren't addressed, including:

1) The buy link text is still incorrect
2) Contributors text style is still not correct. Italics has been removed from all of it, not just the names as requested.
3) There is still too much bottom padding, IMO, on the hero image (too much space below the social icons)
4) The spacing between title, subtitle, authors, and contributors is still horsey. Happy to fine tune this with you at your desk, if needed — but if you compare to the comp, you'll see that the comp is generally tighter here. I'm thinking authors needs to move up a bit, and contributors needs to be a but closer to authors:
image


There's also at least one new bug that I didn't see before:
1) The text on the draft label is no longer visible

**Comment by naomiyaki** [comment]
@zdavis I've force pushed another set of changes:
**Draft label colors:** Fixed to match comps

**Space between title and subtitle:** There's no example of Title + Subtitle in the UI Updates file with the new hero. The spacing was already much less than on the original comp (UI Projects), but I lowered it even more. Please let me know if this works for you.

**Space between subtitle and authors, space between title (without subtitle)  and authors:** Exact to the comps within 1 pixel

**Space between author (no avatars) and contributors**: Exact to the comps

**Space between author (avatars) and contributors:** No design provided so use same spacing as above

**Space between contributors and divider line:** Exact to comps

**Space between authors (with/without avatars) and divider line:** Exact to comps

**Space between authors (with avatars) and contributors:** No example given, so used same spacing as without avatars plus a pixel or two.

**Space between social and hashtag:** Exact to comps

**Bottom padding on hero:** This was exact to comps before, so I lowered it slightly. Please let me know if you want further adjustments.

**Comment by zdavis** [comment]
@naomiyaki 

Just a reminder to avail yourself of Lodash methods to make your code easier to understand. Instead of this:

```
  hasAvatars(creators, variant) {
    let result = true;
    creators.forEach(creator => {
      const hasAvatarVariant = get(
        creator.attributes,
        `avatarStyles.${variant}`
      );
      if (!hasAvatarVariant) {
        result = false;
      }
    });
    return result;
  }
```

Just do this:

```
  import some from "lodash/some";
  import isEmpty from "lodash/isEmpty";

  hasAvatars(creators, variant) {
    return !some(creators, (c) => isEmpty(c.attributes.avatarStyles[variant]));
  }
```

The lodash version has the benefit of being more performant. With a forEach statement, you loop through every item in the collection. With `some`, the loop terminates as soon as it finds an item in the collection for which the predicate returns true. The shorter version is more legible, too, and it's easier to see what's happening.
